### ISSUE TITLE
Store API: Adding crossSellings association on a product where no crossSellings are defined leads to HTTP 500

### ISSUE THREAD

**Issue by juripetersen** [shopware/platform#2156]
### PHP Version

8.0.0

### Shopware Version

6.4.5.0

### Expected behaviour

When requesting a product with crossSellings set, the behaviour works:

POST /store-api/product/${id}

```
{
	"associations": {
		"crossSellings": {}
	}
}
```

```
{
  "apiAlias": "product_detail",
  "product": {
    "crossSellings": [
       {...}
    ]
  },
  "configurator": []
}
```

When requesting a product with no crossSelling products set:

POST /store-api/product/${id}

```
{
	"associations": {
		"crossSellings": {}
	}
}
```

```
{
  "apiAlias": "product_detail",
  "product": {
    "crossSellings": []
  },
  "configurator": []
}
```

### Actual behaviour

POST /store-api/product/${id}

```
{
	"associations": {
		"crossSellings": {}
	}
}

```

```
  "errors": [
    {
      "code": "0",
      "status": "500",
      "title": "Internal Server Error",
      "detail": "Shopware\\Core\\Framework\\DataAbstractionLayer\\EntityCollection::getList(): Argument #1 ($ids) must be of type array, null given, called in \/vendor\/shopware\/core\/Framework\/DataAbstractionLayer\/Dbal\/EntityReader.php on line 547"
    }
  ]
}

```

### How to reproduce

- Create two products. 
- Add a crossSelling product to one.
- Request both via store-api and add a crossSellings association.

**Comment by keulinho** [comment]
Maybe a duplicate of [this issue](https://issues.shopware.com/issues/NEXT-18081).

**Comment by juripetersen** [comment]
This is solved in version 6.4.7.0
